Kubernetes/legos/k8s_remove_pod_from_deployment/k8s_remove_pod_from_deployment.py

```diff
@@ -0,0 +1,97 @@
+
+       :type handle: Object
+       :param handle: Object returned from task.validate(...) routine
+
+       :type pod_name: str
+       :param pod_name: Name of the K8S POD (Mandatory parameter)
+
+       :type namespace: str 
+       :param namespace: Namespace where the above K8S POD is found (Mandatory parameter)
+
+       :rtype: None
+    """
+    if not pod_name or not namespace:
+        raise Exception("Pod Name and Namespace are Mandatory fields")
+
+    core_api = client.CoreV1Api(api_client=handle)
+    apps_api = client.AppsV1Api(api_client=handle)
+
+    # Labels are key-value pairs that can be attached to Kubernetes objects. 
+    # Labels can be used to organize and group objects, and they can be used to 
+    # select objects for operations such as deletion and updates.
+
+    # Selectors are used to select a group of objects for an operation. Selectors can be 
+    # specified using labels, and they can be used to select all objects with a given 
+    # label or all objects that match a certain pattern.
+
+    # Kubernetes deployment uses Labels and Selectors to select which pods need to be 
+    # updated when a new version of a pod is deployed.
+
+    # Here by modifying the selector label for deployment, we are making sure the pod
+    # is removed from the deployment. We verify the same by listing the pod labels after
+    # doing a patch operation 
+    try:
+        pod = core_api.read_namespaced_pod(name=pod_name, namespace=namespace)
+        owner_references = pod.metadata.owner_references
+        deployment_name = ''
+        if isinstance(owner_references, list):
+            owner_name = owner_references[0].name
+            owner_kind = owner_references[0].kind 
+            if owner_kind == 'Deployment':
+                deployment_name = owner_name 
+
+        if deployment_name != '':
+            deployment = apps_api.read_namespaced_deployment(name=deployment_name, namespace=namespace)
+            deployment_labels  = [key for key, value in deployment.spec.selector.match_labels.items()]
+    
+            pod_labels = [key for key,value in pod.metadata.labels.items()]
+
+            common_labels = set(deployment_labels) & set(pod_labels)
+            new_label = {}
+            for label in common_labels:
+                new_label[label] = pod.metadata.labels.get(label) + '-out-for-maintenance'
+
+            pod.metadata.labels.update(new_label)
+            core_api.patch_namespaced_pod(pod_name, namespace, pod)
+
+            label_selector = ','.join([f'{key}={value}' for key, value in deployment.spec.selector.match_labels.items()])
+            updated_pods = core_api.list_namespaced_pod(namespace, label_selector=label_selector)
+            updated_pod_names = [x.metadata.name for x in updated_pods.items]
+            if pod_name not in updated_pod_names:
+                print(f"Successfully Removed POD from {deployment.metadata.name} {pod_name} in {namespace} Namespace")
```

Comment:
Can we be little bit more cautious about how are we printing, i see capital letters in the middle of sentence. This should be 
```
Successfully removed pod {pod_name} from deployment {deployment.metadata.name} in namespace {namespace}
```

Reply:
Removed the check itself.